src/main/kotlin/edu/carroll/polyominoes/service/account/UserDetailsServiceImpl.kt

```diff
@@ -0,0 +1,64 @@
+package edu.carroll.polyominoes.service.account
+
+import edu.carroll.polyominoes.jpa.repo.AccountRepository
+import edu.carroll.polyominoes.service.account.model.SecurityAccount
+import org.slf4j.LoggerFactory
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import org.springframework.stereotype.Service
+
+@Service
+class UserDetailsServiceImpl(private val accountRepo: AccountRepository) : UserDetailsService {
+    companion object {
+        private val log = LoggerFactory.getLogger(UserDetailsServiceImpl::class.java)
+    }
+
+
+    /**
+     * Locates the user based on the username. In the actual implementation, the search
+     * may possibly be case sensitive, or case insensitive depending on how the
+     * implementation instance is configured. In this case, the `UserDetails`
+     * object that comes back may have a username that is of a different case than what
+     * was actually requested..
+     * @param username the username identifying the user whose data is required.
+     * @return a fully populated user record (never `null`)
+     * @throws UsernameNotFoundException if the user could not be found or the user has no
+     * GrantedAuthority
+     */
+    @Throws(UsernameNotFoundException::class)
+    override fun loadUserByUsername(username: String?): UserDetails {
+
+        if (username.isNullOrBlank()) {
+            log.warn("loadUserByUsername: username was null or blank")
+            throw UsernameNotFoundException("username was null or blank")
+        }
+
+        log.debug("loadUserByUsername: attempting to find user '{}'", username)
+
+        val users = if (username.contains("@")) {
+            log.debug("loadUserByUsername: username contained '@' looking up '{}' by email", username)
+            // Always do the lookup in a case-insensitive manner (lower-casing the data).
```

Comment:
The javadoc above imply the details are unknown with the case sensitive/insensitive lookups, but the actual implementation here show it's case insensitive.